### PR TITLE
New API proposal - High-throughput elastic networks API

### LINKED ISSUE #89: IP high-throughput elastic network API
Proposal to create an "IP high-throughput elastic network" API:
This API provides scheduling ability of huge data transmission in network. 
It can calculate load-balancing paths automatically with given time duration, total data amount and bandwidth limit, and satisfy huge data transmission requirements with dynamic and elastic bandwidth allocation algorithms, tidal-traffic-aware segment list adjustment and SRv6 multiple segment list technology. 
With this API, developers can monetize their dynamic bandwidth requirements effectively.
Use cases: technological research, astronomical and meteorological analysis, data backup, video editing, and so on.

### PR DESCRIPTION
**Pull request by wangruobei**
#### What type of PR is this?

New API proposal 'High-throughput elastic networks API'

#### What this PR does / why we need it:
'IP high-throughput elastic network API' provides the scheduling ability of huge data transmission in the network.

#### Which issue(s) this PR fixes:

Fixes #89

### CODE REVIEW

File: documentation/API proposals/APIproposal_IP-high-throughput-elastic-network_ChinaMobile.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
...Cross-cloud backup, model training at intelligent compute centers, video editing, and scientific computing. For example, the Five-hundred-meter Aperture Spherical radio Telescope (FAST), located in GuiZhou province of China, could generate more than 10 PB of observational data per year. By calling this API, users can dynamically adjust bandwidth and elastically orchestrate network resources according to their specific requirements for bandwidth, latency, and file size. This capability facilitates the efficient and expedient transmission of large observational datasets within short timeframes. <br /><br />Input:<br/>1. Bandwidth Requirement: Time-segmented bandwidth requirements (can include multiple time points)<br/>2. Bandwidth Requirement Type: Upload bandwidth, download bandwidth, two-way bandwidth Service<br/>3. Latency Requirement: Latency guarantee requirement range<br/>4. File Requirements: Size of files to be transmitted <br/>5. Guarantee Requirements: Bandwidth guarantee and load sharing requirements<br/><br />Output:<br/>1. Estimation of transmission rate by time segment<br/>2. Estimated transmission duration,<br/>3. Statistical conclusions after successful transmission |
```
Comment (Kevsy) [inline documentation/API proposals/APIproposal_IP-high-throughput-elastic-network_ChinaMobile.md:7]: The API Summary states : "This API provides **scheduling** ability of huge data transmission in network.". 

So I think the primary operation is a request to schedule a huge data transmission, based on the bandwidth/latency/loss etc. input parameters. 

And I think the output response should include:
1) an indication that the operator is able to meet the scheduling request (or not).
2) a schedule of date(s) showing when the operator can achieve the huge data transmission (e.g. an array of network quiet times) . I think the 'estimation of transmission rate', 'estimated transmission duration' are specific to each individual item in that array (i.e. each possible schedule time that the operator can offer). The API Consumer may choose the schedule date with the best performance, or they could choose a schedule date with inferior performance which fulfils the transmission requirements but with less bandwidth than the best schedule date. For example, the API Consumer may require the earliest possible date, or the cheapest option.
3) Any cost difference between the different schedule dates (if appropriate).
4) (optional) any conditions required for the operator to meet the request (e.g. a particular entry point to the network must be used).
5) Statistical conclusions after successful **and unsuccessful** transmission.

_(note this captures my comment from the API Backlog meeting 10-10-24)_

Comment (wangruobei) [reply documentation/API proposals/APIproposal_IP-high-throughput-elastic-network_ChinaMobile.md]: More information on technology dependency: 
The IP high-throughput elastic network API relies on an SRv6-based IP bearer network, and this IP bearer network:
1.	Is based on the Segment Routing Architecture (IETF RFC8402);
2.	Implements SR multi-Segment load-sharing technology through SR v6 network programming (IETF RFC8986);
3.	Adjusts QoS in real-time using the Network Configuration Protocol (NETCONF) (IETF RFC6241) to ensure the availability of the high-throughput large bandwidth capability API.

Differences from current proposals:
IP High-Throughput Elastic Network API: The IP high-throughput elastic network API relies on an IPSR v6 network and achieves the capability of huge data transmission in the network through srv6 technology, such as UCMP or ECMP. It is a technology within the domain of IP bearer networks and thus differs significantly in terms of network domain and business functionality from the Dedicated Networks API (5G NPN), The QoD API (Home Devices), and the Network Slicing API (RAN slicing).
Dedicated Networks API: This API relies on networking features such as the 5G NPN (specifically the PNI-NPN, TS 23.501 Clause 5.30) or the Network Communication Service (GSMA OPG.02 Annex H), which offers several realization options including Slices or APN/DNN.
The QoD API (Home Devices Quality on Demand (QoD)): This API necessitates the support of Differentiated Services (DiffServ) on home network routers as a mechanism for classifying and managing network traffic and providing quality of service (QoS).
Network Slicing API: This API requires the availability of RAN slicing, UDM, PCF, and network slicing management services as specified in 3GPP TS 28.531 (network slice provisioning operations), TS 28.541 (Information model definitions for network slice NRM) standards.

**Comment by eric-murray** [comment]
@wangruobei
Thank you for updating the proposal. Can you also provide your supporters for this API? There is a row in the template for this information.

cc @jgarciahospital

**Comment by jgarciahospital** [comment]
Thanks for the update. 

Please include the list of companies supporting as mentioned by @eric-murray and also you'll need to solve the easyCLA issue (follow https://github.com/camaraproject/EasyCLA or contact LF support @wrathwolf)

**Comment by wangruobei** [comment]
@eric-murray 
Thank you for your feedback. I am pleased to inform you that we have secured support from our company(**China Mobile**) as well as from **Huawei** for this API. Additionally, the API proposal has already been presented at the OGW meeting. We are unsure if this presentation qualifies as obtaining OGW's official support, and would appreciate further clarification on this matter.

**Review by jgarciahospital — APPROVED** [review]

**Pull request merged**